app/src/main/java/com/keylesspalace/tusky/components/announcements/AnnouncementAdapter.kt

```diff
@@ -0,0 +1,115 @@
+/* Copyright 2020 Tusky Contributors
+ *
+ * This file is a part of Tusky.
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Tusky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Tusky; if not,
+ * see <http://www.gnu.org/licenses>. */
+
+package com.keylesspalace.tusky.components.announcements
+
+import android.view.ContextThemeWrapper
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.core.view.size
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.material.chip.Chip
+import com.google.android.material.chip.ChipGroup
+import com.keylesspalace.tusky.R
+import com.keylesspalace.tusky.entity.Announcement
+import com.keylesspalace.tusky.entity.Emoji
+import com.keylesspalace.tusky.util.emojify
+import kotlinx.android.synthetic.main.item_announcement.view.*
+
+interface AnnouncementActionListener {
+    fun openReactionPicker(announcementId: String, target: View)
+    fun addReaction(announcementId: String, name: String)
+    fun removeReaction(announcementId: String, name: String)
+}
+
+class AnnouncementAdapter(
+        private var items: List<Announcement>,
```

Comment:
I would rather put `var items: List<Announcement> = emptyList()` into the class body